tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php

```diff
@@ -127,15 +127,33 @@ public function shouldPassOnCorrectSequenceOfUsersWhenMerging(): void
     }
 
     /**
-     * ([interaction], [user merge], [recommendation]): (A, B -> A, A)
+     * ([interaction], [user merge], [recommendation]): (B, A -> B, B)
+     *
+     * @test
+     */
+    public function shouldPassOnCorrectSequenceOfUsersWhenMergingWithInteractionUserSameAsRecommendation(): void
```

Comment:
Maybe there is opportunity for data provider ($interaction, $userMerge, $recommendation) and just extend the previous testcase, what do you think?

Reply:
ok, changed to use same test with dataprovider